python_ta/cfg/visitor.py

```diff
@@ -312,11 +323,13 @@ def _visit_jump(
                 self._current_cfg.link(old_curr, exits[type(node).__name__])
                 old_curr.add_statement(node)
                 break
+
         else:
-            raise SyntaxError(
+            logging.warning(
                 f"'{type(node).__name__}' outside"
                 f' {"function" if isinstance(node, nodes.Return) else "loop"}'
             )
+
         unreachable_block = self._current_cfg.create_block()
```

Comment:
@MariaMa-GitHub okay I looked into this a bit more. You've made good progress, and the issue with the "black" box not being correct linked is due to the following reason. With the "valid" jumps inside the loop, the logic has two steps:

1. link `old_curr` with the correct "exit block"
2. add `node` to `old_curr` (so that it appears inside the block). This needs to happen _after_ the linking because the current implementation of `link` restricts linking when the last statement inside the block is a jump node. (I suspect this is actually not the best design, but it's what we have right now.)

So to mimic this behaviour for the "invalid" jump cases, you need to (1) link the current block to the new `unreachable` block created at the bottom of this file, and then (2) call `add_statement` directly, mimicking the "valid case" logic. `visit_generic` should also work, but I think it's best to be consistent with the existing code.